modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java

```diff
@@ -390,6 +392,8 @@ private void flattenComposedChildren(OpenAPI openAPI, String key, List<Schema> c
                         listIterator.set(schema);
                     }
                 }
+            } else {
+                // likely a reference to schema (not inline schema)
```

Comment:
Maybe we should log something here?

Have you tested with an object without type or properties defined (additionalProperties only)?

Reply:
I tried to log something here - it's just a reference to a model. (e.g. Foo), which means it's not an inline schema.

Reply:
> Have you tested with an object without type

Yes of course. Let me add a test case tomorrow so that we'll be alerted in case the behavior has changed.

Reply:
Will add test cases in a separate PR.

Reply:
Added a test via https://github.com/OpenAPITools/openapi-generator/pull/6042